diff --git a/mm/2s2h/BenGui/BenMenu.cpp b/mm/2s2h/BenGui/BenMenu.cpp
index f84cd6426..b16c4224f 100644
--- a/mm/2s2h/BenGui/BenMenu.cpp
+++ b/mm/2s2h/BenGui/BenMenu.cpp
@@ -683,6 +683,9 @@ void BenMenu::AddEnhancements() {
     AddWidget(path, "Prevent Diving Over Water", WIDGET_CVAR_CHECKBOX)
         .CVar("gEnhancements.Player.PreventDiveOverWater")
         .Options(CheckboxOptions().Tooltip("Prevents Link from automatically diving over bodies of water."));
+    AddWidget(path, "Underwater Ocarina", WIDGET_CVAR_CHECKBOX)
+        .CVar("gEnhancements.Player.UnderwaterOcarina")
+        .Options(CheckboxOptions().Tooltip("Allows Zora to use the Ocarina of Time when grounded underwater."));
     AddWidget(path, "Manual Jump", WIDGET_CVAR_CHECKBOX)
         .CVar("gEnhancements.Player.ManualJump")
         .Options(CheckboxOptions().Tooltip("Z + A to Jump and B while midair to Jump Attack."));
diff --git a/mm/2s2h/Enhancements/Player/UnderwaterOcarina.cpp b/mm/2s2h/Enhancements/Player/UnderwaterOcarina.cpp
new file mode 100644
index 000000000..b6ff81fdc
--- /dev/null
+++ b/mm/2s2h/Enhancements/Player/UnderwaterOcarina.cpp
@@ -0,0 +1,21 @@
+#include "public/bridge/consolevariablebridge.h"
+#include "2s2h/GameInteractor/GameInteractor.h"
+#include "2s2h/ShipInit.hpp"
+
+extern "C" {
+#include "z64item.h"
+}
+
+#define CVAR_NAME "gEnhancements.Player.UnderwaterOcarina"
+#define CVAR CVarGetInteger(CVAR_NAME, 0)
+
+void RegisterUnderwaterOcarina() {
+    COND_VB_SHOULD(VB_DISABLE_ITEM_UNDERWATER_FLOOR, CVAR, {
+        const auto item = va_arg(args, s32);
+        if (item == ITEM_OCARINA_OF_TIME) {
+            *should = false;
+        }
+    });
+}
+
+static RegisterShipInitFunc initFunc(RegisterUnderwaterOcarina, { CVAR_NAME });
diff --git a/mm/2s2h/GameInteractor/GameInteractor.h b/mm/2s2h/GameInteractor/GameInteractor.h
index 8240242a6..9b8acbeb4 100644
--- a/mm/2s2h/GameInteractor/GameInteractor.h
+++ b/mm/2s2h/GameInteractor/GameInteractor.h
@@ -239,6 +239,8 @@ typedef enum {
     VB_SETUP_TRANSITION,
     VB_BE_NEAR_DOOR,
     VB_LOAD_PLAYER_ANIMATION_FRAME,
+    // Vanilla Condition: !((GET_CUR_FORM_BTN_ITEM(i) >= ITEM_BOTTLE) && (GET_CUR_FORM_BTN_ITEM(i) <= ITEM_OBABA_DRINK))
+    VB_DISABLE_ITEM_UNDERWATER_FLOOR,
 } GIVanillaBehavior;
 
 typedef enum {
diff --git a/mm/2s2h/PresetManager/PresetManager.cpp b/mm/2s2h/PresetManager/PresetManager.cpp
index 62b1fc441..e4efda914 100644
--- a/mm/2s2h/PresetManager/PresetManager.cpp
+++ b/mm/2s2h/PresetManager/PresetManager.cpp
@@ -147,7 +147,8 @@ nlohmann::json curatedPresetJ = R"(
                 "FierceDeityPutaway": 1,
                 "InfiniteDekuHopping": 1,
                 "InstantPutaway": 1,
-                "PreventDiveOverWater": 1
+                "PreventDiveOverWater": 1,
+                "UnderwaterOcarina": 1
             },
             "PlayerActions": {
                 "ArrowCycle": 1,
diff --git a/mm/src/code/z_parameter.c b/mm/src/code/z_parameter.c
index 8dec3f0b6..86d716093 100644
--- a/mm/src/code/z_parameter.c
+++ b/mm/src/code/z_parameter.c
@@ -2585,8 +2585,10 @@ void Interface_UpdateButtonsPart2(PlayState* play) {
         for (i = EQUIP_SLOT_C_LEFT; i <= EQUIP_SLOT_C_RIGHT; i++) {
             if (GET_CUR_FORM_BTN_ITEM(i) != ITEM_MASK_ZORA) {
                 if (Player_GetEnvironmentalHazard(play) == PLAYER_ENV_HAZARD_UNDERWATER_FLOOR) {
-                    if (!((GET_CUR_FORM_BTN_ITEM(i) >= ITEM_BOTTLE) &&
-                          (GET_CUR_FORM_BTN_ITEM(i) <= ITEM_OBABA_DRINK))) {
+                    if (GameInteractor_Should(VB_DISABLE_ITEM_UNDERWATER_FLOOR,
+                                              !((GET_CUR_FORM_BTN_ITEM(i) >= ITEM_BOTTLE) &&
+                                                (GET_CUR_FORM_BTN_ITEM(i) <= ITEM_OBABA_DRINK)),
+                                              (s32)GET_CUR_FORM_BTN_ITEM(i))) {
                         if (gSaveContext.buttonStatus[i] == BTN_ENABLED) {
                             restoreHudVisibility = true;
                         }
@@ -2612,8 +2614,10 @@ void Interface_UpdateButtonsPart2(PlayState* play) {
         for (s16 j = EQUIP_SLOT_D_RIGHT; j <= EQUIP_SLOT_D_UP; j++) {
             if (DPAD_GET_CUR_FORM_BTN_ITEM(j) != ITEM_MASK_ZORA) {
                 if (Player_GetEnvironmentalHazard(play) == PLAYER_ENV_HAZARD_UNDERWATER_FLOOR) {
-                    if (!((DPAD_GET_CUR_FORM_BTN_ITEM(j) >= ITEM_BOTTLE) &&
-                          (DPAD_GET_CUR_FORM_BTN_ITEM(j) <= ITEM_OBABA_DRINK))) {
+                    if (GameInteractor_Should(VB_DISABLE_ITEM_UNDERWATER_FLOOR,
+                                              !((DPAD_GET_CUR_FORM_BTN_ITEM(j) >= ITEM_BOTTLE) &&
+                                                (DPAD_GET_CUR_FORM_BTN_ITEM(j) <= ITEM_OBABA_DRINK)),
+                                              (s32)DPAD_GET_CUR_FORM_BTN_ITEM(j))) {
                         if (gSaveContext.shipSaveContext.dpad.status[j] == BTN_ENABLED) {
                             restoreHudVisibility = true;
                         }
