--- a/Utils.py
+++ b/Utils.py
@@ -198,11 +198,17 @@
                     not filecmp.cmp(local_path("manifest.json"), user_path("manifest.json"), shallow=True):
                 import shutil
                 for dn in ("Players", "data/sprites", "data/lua"):
-                    shutil.copytree(local_path(dn), user_path(dn), dirs_exist_ok=True)
+                    for src_dir, _, file_names in os.walk(local_path(dn)):
+                        dst_dir = os.path.join(user_path(dn), os.path.relpath(src_dir, local_path(dn)))
+                        os.makedirs(dst_dir, exist_ok=True)
+                        for file_name in file_names:
+                            shutil.copyfile(os.path.join(src_dir, file_name), os.path.join(dst_dir, file_name))
                 if not os.path.exists(local_path("manifest.json")):
                     warnings.warn(f"Upgrading {user_path()} from something that is not a proper install")
                 else:
-                    shutil.copy2(local_path("manifest.json"), user_path("manifest.json"))
+                    shutil.copyfile(local_path("manifest.json"), user_path("manifest.json"))
+                    manifest_stat = os.stat(local_path("manifest.json"))
+                    os.utime(user_path("manifest.json"), ns=(manifest_stat.st_atime_ns, manifest_stat.st_mtime_ns))
             os.makedirs(user_path("worlds"), exist_ok=True)
 
     return os.path.join(user_path.cached_path, *path)
