diff --git a/soh/soh/OTRGlobals.h b/soh/soh/OTRGlobals.h
index 4dcb2f545..a307a38c8 100644
--- a/soh/soh/OTRGlobals.h
+++ b/soh/soh/OTRGlobals.h
@@ -31,7 +31,7 @@ extern std::unordered_map<std::string, ExtensionEntry> ExtensionCache;
 #include "Enhancements/randomizer/settings.h"
 
 const std::string customMessageTableID = "BaseGameOverrides";
-const std::string appShortName = "soh";
+const std::string appShortName = "soh-anchor";
 
 #ifdef __WIIU__
 const uint32_t defaultImGuiScale = 3;
