diff --git a/src/backend/render/element.rs b/src/backend/render/element.rs
index e1259dc..92030e7 100644
--- a/src/backend/render/element.rs
+++ b/src/backend/render/element.rs
@@ -189,19 +189,7 @@ where
         }
     }
 
-    fn kind(&self) -> Kind {
-        match self {
-            CosmicElement::Workspace(elem) => elem.kind(),
-            CosmicElement::Cursor(elem) => elem.kind(),
-            CosmicElement::Dnd(elem) => elem.kind(),
-            CosmicElement::MoveGrab(elem) => elem.kind(),
-            CosmicElement::Postprocess(elem) => elem.kind(),
-            CosmicElement::Zoom(elem) => elem.kind(),
-            CosmicElement::Damage(elem) => elem.kind(),
-            #[cfg(feature = "debug")]
-            CosmicElement::Egui(elem) => elem.kind(),
-        }
-    }
+    fn kind(&self) -> Kind { Kind::Unspecified }
 
     fn is_framebuffer_effect(&self) -> bool {
         match self {
